lib/bdev/bdev.c

```diff
@@ -7796,12 +7796,14 @@ bdev_open(struct spdk_bdev *bdev, bool write, struct spdk_bdev_desc *desc)
 	}
 
 	if (write &&
-	    bdev->internal.claim_type == SPDK_BDEV_CLAIM_EXCL_WRITE &&
-	    bdev->internal.claim.v1.module &&
-	    strcmp(bdev->internal.claim.v1.module->name, "NVMe-oF Target")) {
-		LOG_ALREADY_CLAIMED_ERROR("already claimed", bdev);
-		spdk_spin_unlock(&bdev->internal.spinlock);
-		return -EPERM;
+	    bdev->internal.claim_type != SPDK_BDEV_CLAIM_NONE) {
```

Comment:
For what scenario, this code change is done ? The new  condition is opening more window for the `already claim` scenario.

Reply:
The previous fix was failing unittests and triggering ASAN:
1. it did not fail claim != none if not nvme-tcp
2. it dereferenced pointer whithouht checking for null